include/revng/Model/BinaryDiff.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include "revng/Model/Binary.h"
+#include "revng/Model/Generated/AllTypesVariant.h"
+#include "revng/TupleTree/TupleTreeDiff.h"
+
+template<>
+struct TupleTreeEntries<model::Binary> {
+  using Types = model::AllTypes;
+};
```

Comment:
Uhm, we can generate this. In fact, instead of `AllTypes` we could just specialize the `TupleTreeEntries` trait.

Reply:
i tried, but importing tupleTreeDiff.h not as last headers among the generated one broke everything and i have been unable to figure out what the issue was. I think it has something to do with template declaration order, i can put TupleTreeEntries in a stand alone header toh.

Reply:
Works. What matters is that we generate stuff and we don't have to do this for every user of `TupleTreeGenerator`.